engine/src/main/java/com/datatorrent/stram/engine/OperatorContext.java

```diff
@@ -84,12 +87,13 @@ public void setIdleTimeout(long idleTimeout)
    * @param attributes the value of attributes
    * @param parentContext
    */
-  public OperatorContext(int id, AttributeMap attributes, Context parentContext)
+  public OperatorContext(int id, String operatorName, AttributeMap attributes, Context parentContext)
   {
     super(attributes, parentContext);
     this.lastProcessedWindowId = Stateless.WINDOW_ID;
     this.id = id;
     this.stateless = super.getValue(OperatorContext.STATELESS);
+    this.operatorName = Preconditions.checkNotNull(operatorName, "operator name");
```

Comment:
I tried to call `LogicalPlan.addOperator(null, GenericTestOperator.class)` in LogicalPlanTest and no exception was raised until `LogicalPlan.validate()` was called. IMO, both `null` and empty string should be disallowed in the addOperator() itself and OperatorContext should not do additional checks. For example, OperatorMeta does not do such checks.